exercises/binary-search/example.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stdexcept>
+#include "binary_search.h"
+
+namespace binary_search {
+
+size_t find(std::vector<int> const& data, int const value)
+{
+    if(data.empty() || value < data.front() || data.back() < value)    
```

Comment:
A idea on style - some prefer to extract a complex boolean into a helper function such as:
```c++
if(!is_value_in_range(data, value))
```
feeling that it adds expressiveness to the code. Others don't like the extra level of indirection. I don't know if exercism has a preference.

It works nicely as it is; just food for thought.

Reply:
Good point, thanks @KevinWMatthews. I introduced a function, as per your suggestion.